src/main/java/org/homework/repository/TodoRepository.java

```diff
@@ -2,42 +2,59 @@
 
 import org.homework.domain.Todo;
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Objects;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public class TodoRepository {
 
     private static int seq = 0;
     private static final Map<Integer, Todo> todoMap = new HashMap<>();
 
-    public static boolean containsTodo(int todoId) {
-        return todoMap.containsKey(todoId);
-    }
-
-    public static int addTodo(String content) {
-        Todo todo = new Todo(++seq, content);
+    public static int addTodo(String content, LocalDate endDate) {
+        Todo todo = new Todo(++seq, content, endDate);
 
         todoMap.put(todo.getId(), todo);
         return todo.getId();
     }
 
-    public static Todo getTodo(int todoId) {
-        if (todoMap.containsKey(todoId)) {
-            return todoMap.get(todoId);
-        }
+    public static Optional<Todo> getTodo(int todoId) {
+        return Optional.ofNullable(todoMap.get(todoId));
+    }
+
+    public static List<Todo> getWeekTodoList() {
+        LocalDate today = LocalDate.now();
 
-        return null;
+        return new ArrayList<>(todoMap.values()).stream()
+                .filter(todo -> {
+                    long betweenDays = today.until(todo.getDeadline(), ChronoUnit.DAYS);
+                    return betweenDays >= 0 && betweenDays <= 7;
+                })
+                .sorted(Comparator.comparing(Todo::getDeadline))
+                .collect(Collectors.toList());
     }
 
+    public static List<Todo> getSearchTodoList(String keyword) {
+        return new ArrayList<>(todoMap.values()).stream()
```

Comment:
```suggestion
        return todoMap.values().stream()
```
ArrayList로 선언하지 않고 바로 사용할수있습니다.